autotest/autotest_services/tests/wcs/test_v20.py

```diff
@@ -1632,4 +1632,79 @@ def getExpectedHTTPStatus(self):
         return 404
 
     def getExpectedExceptionCode(self):
-        return "TilingInvalid"
\ No newline at end of file
+        return "TilingInvalid"
+
+
+@tag('wcs', 'wcs20')
+class WCS20DefaultErrorFormatIsXmlTestCase(testbase.OWSTestCase):
+    def getRequest(self):
+        params = "service=WCS&version=2.0.1&request=invalid"
+        return (params, "kvp")
+
+    def testStatus(self):
+        pass
+
+    def testContentTypeIsXml(self):
+        content_type = self.response.get("Content-Type")
+        self.assertEqual(content_type, "text/xml")
+
+
+@tag('wcs', 'wcs20')
+class WCS20ErrorFormatIsHtmlOnRequestTestCase(testbase.OWSTestCase):
+    def getRequest(self):
+        params = "service=WCS&version=2.0.1&request=invalid&exceptions=text/html"
+        return (params, "kvp")
+
+    def testStatus(self):
+        pass
+
+    def testContentTypeIsHtml(self):
+        content_type = self.response.get("Content-Type")
+        self.assertEqual(content_type, "text/html")
+
+    def testTemplateContainsErrorMessage(self):
+        self.assertIn(
+            "Error: Operation &#39;INVALID&#39; is not supported",
+            self.response.content.decode(),
+        )
+
+@tag('wcs', 'wcs20')
+class WCS20PostDefaultErrorFormatIsXmlTestCase(testbase.OWSTestCase):
+    def getRequest(self):
+        params = """<ns:invalid updateSequence="u2001" service="WCS"
+          xmlns:ns="http://www.opengis.net/wcs/2.0"
+          xmlns:ns1="http://www.opengis.net/ows/2.0">
+            <ns1:AcceptVersions><ns1:Version>2.0.1</ns1:Version></ns1:AcceptVersions>
+          </ns:invalid>
+        """
+        return (params, "xml")
+
+    def testStatus(self):
+        pass
+
+    def testContentTypeIsHtml(self):
+        content_type = self.response.get("Content-Type")
+        self.assertEqual(content_type, "text/xml")
+
+
+@tag('wcs', 'wcs20')
+class WCS20PostErrorFormatIsHtmlOnRequestTestCase(testbase.OWSTestCase):
+    def getRequest(self):
+        params = """<ns:invalid updateSequence="u2001" service="WCS"
+          xmlns:ns="http://www.opengis.net/wcs/2.0"
```

Comment:
It is strange to see `ns` as the namespace prefix, but technically correct. It would be more readable to use `wcs` and `ows` instead.

Reply:
I agree, this definitely wasn't intended like this.

Btw I'm not sure how you want to handle such things, but in the spirit of collective code ownership I would be ok with reviewers implementing changes of this size directly.

Reply:
Yes, I would agree as well